qa/rgw/store/sfs/tests/test-sfs-lifecycle-smoke.py

```diff
@@ -0,0 +1,276 @@
+        #                   3rd cycle will be fully expired
+        time.sleep(3 * LifecycleSmokeTests.LIFECYCLE_DEBUG_INTERVAL)
+        response = self.s3_client.list_objects(Bucket=bucket_name)
+        self.check_objects_exist(
+            response["Contents"],
+            ["keep2/foo", "keep2/bar", "expire3/foo", "expire3/bar"],
+        )
+
+        time.sleep(LifecycleSmokeTests.LIFECYCLE_DEBUG_INTERVAL)
+        # at this point there are still not enough cycles to count 5 days
+        response = self.s3_client.list_objects(Bucket=bucket_name)
+        self.check_objects_exist(
+            response["Contents"],
+            ["keep2/foo", "keep2/bar", "expire3/foo", "expire3/bar"],
+        )
+
+        # same logic as above for number of cycles plus 1 extra cycle to ensure
+        # bucket was processed
+        time.sleep(4 * LifecycleSmokeTests.LIFECYCLE_DEBUG_INTERVAL)
+        response = self.s3_client.list_objects(Bucket=bucket_name)
+        self.check_objects_exist(response["Contents"], ["keep2/foo", "keep2/bar"])
+
+    def test_lifecycle_versioning_enabled(self):
+        bucket_name = self.get_random_bucket_name()
+        self.s3_client.create_bucket(Bucket=bucket_name)
+        self.assert_bucket_exists(bucket_name)
+        response = self.s3_client.put_bucket_versioning(
+            Bucket=bucket_name,
+            VersioningConfiguration={"MFADelete": "Disabled", "Status": "Enabled"},
+        )
+        test_file_path_1 = os.path.join(self.test_dir.name, "test_file_1.bin")
+        self.generate_random_file(test_file_path_1)
+        # upload the file
+        self.s3_client.upload_file(test_file_path_1, bucket_name, "expire1/test")
+        # now upload again with different content
+        test_file_path_2 = os.path.join(self.test_dir.name, "test_file_2.bin")
+        self.generate_random_file(test_file_path_2)
+        self.s3_client.upload_file(test_file_path_2, bucket_name, "expire1/test")
+        response = self.s3_client.list_object_versions(Bucket=bucket_name, Prefix="")
+        self.assertTrue("Versions" in response)
+        self.assertEqual(2, len(response["Versions"]))
+        self.assertFalse("DeleteMarkers" in response)
+
+        rules = [
+            {
+                "ID": "rule1",
+                "Expiration": {"Days": 1},
+                "Prefix": "expire1/",
+                "Status": "Enabled",
+            }
+        ]
+        lifecycle = {"Rules": rules}
+        self.s3_client.put_bucket_lifecycle_configuration(
+            Bucket=bucket_name, LifecycleConfiguration=lifecycle
+        )
+
+        # give enough time to expire.
+        # 4 cycles because:
+        #                   1st cycle won't be expired yet (not still 1 day)
+        #                   2nd cycle rgw considers the bucket at processed
+        #                       today and skips it
+        #                   3rd cycle will be fully expired
+        #                   4rd cycle is extra
```

Comment:
maybe write down why the extra cycle?

Reply:
changed and now it does not have the extra cycle